src/main/java/ch/cern/eam/wshub/core/services/equipment/impl/LocationServiceImpl.java

```diff
@@ -109,12 +111,8 @@ public String createLocation(InforContext context, Location locationParam) throw
 		locationInfor.getLOCATIONID().setLOCATIONCODE(locationParam.getCode());
 		locationInfor.getLOCATIONID().setORGANIZATIONID(tools.getOrganization(context));
 
-		// Check Custom fields. If they change, or now we have them
-		if (locationParam.getClassCode() != null && (locationInfor.getCLASSID() == null
-				|| !locationParam.getClassCode().toUpperCase().equals(locationInfor.getCLASSID().getCLASSCODE()))) {
-			locationInfor.setUSERDEFINEDAREA(
-					tools.getCustomFieldsTools().getInforCustomFields(context, "LOC", locationParam.getClassCode().toUpperCase()));
-		}
+		USERDEFINEDAREA customFields = getInforCustomFields(context, locationParam, locationInfor);
```

Comment:
Older bug here, but if a class is changed to another that contains the same custom fields, its values are lost